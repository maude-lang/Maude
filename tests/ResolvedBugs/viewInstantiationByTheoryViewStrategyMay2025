#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude -no-prelude \
  < $srcdir/viewInstantiationByTheoryViewStrategyMay2025.maude -no-banner -no-advise \
  > viewInstantiationByTheoryViewStrategyMay2025.out 2>&1

diff $srcdir/viewInstantiationByTheoryViewStrategyMay2025.expected viewInstantiationByTheoryViewStrategyMay2025.out > /dev/null 2>&1
