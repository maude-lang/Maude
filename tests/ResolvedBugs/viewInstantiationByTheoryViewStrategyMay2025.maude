***
*** Counterpart of the viewInstantiationByTheoryViewApril2025 test for
*** strategy to strategy expression mappings.
***

sth T is
  sort Small .
  strat f : Small @ Small .
endsth

sth T2 is
  sort Little .
  strat h : Little @ Little .
endsth

view TV from T to T2 is
  sort Small to Little .
  strat f to h .
endv

sth B is
  sort Tiny .
  strat g : Tiny @ Tiny .
endsth

smod USE-B{X :: B} is
  sd g(N:X$Tiny) := match N:X$Tiny [label "raw thumb"] .
endsm

smod BASE{X :: T} is
  sort Big .
  subsort X$Small < Big .  *** this triggers the bug
endsm

view PV{X :: T} from B to BASE{X} is
  sort Tiny to X$Small .
  strat g(N:Tiny) to expr f(N:X$Small) .
endv

smod TEST{Y :: T2} is
  pr USE-B{PV{TV}{Y}} .
endsm

show sds .
