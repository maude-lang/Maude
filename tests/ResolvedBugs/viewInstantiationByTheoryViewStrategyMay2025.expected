sd h(N:Y$Little) := match N:Y$Little .
Bye.
